packages/grafana-ui/src/components/uPlot/config.ts

```diff
@@ -0,0 +1,75 @@
+import { SelectableValue } from '@grafana/data';
+
+export enum AxisPlacement {
+  Auto = 'auto', // First axis on the left, the rest on the right
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+  Left = 'left',
+  Hide = 'hide',
+}
+
+export function getUPlotSideFromAxis(axis: AxisPlacement) {
+  switch (axis) {
+    case AxisPlacement.Top:
+      return 0;
+    case AxisPlacement.Right:
+      return 1;
+    case AxisPlacement.Bottom:
+      return 2;
+    case AxisPlacement.Left:
+  }
+  return 3; // default everythign to the left
+}
+
+export enum PointMode {
+  Auto = 'auto', // will show points when the density is low or line is hidden
+  Always = 'always',
+  Never = 'never',
+}
+
+export enum GraphMode {
+  Line = 'line', // default
+  Bar = 'bar', // will also have a gap percent
+  Points = 'points', // use the points config
+  Staircase = 'staircase', // https://leeoniya.github.io/uPlot/demos/line-stepped.html
+  Smooth = 'smooth', // https://leeoniya.github.io/uPlot/demos/line-smoothing.html
```

Comment:
After thinking more on merging the staircase and smooth with the mode here not sure. 

I can see a future problem with this where we want to support more line interpolation modes than just staircase & smooth (for example step before step after, smooth has different algos as well). 

Another thing that feels a bit awkward is having two Points modes. GraphMode.Points and a PointsMode. Think we should hide PointsMode if GraphMode = Points (as PointsMode would not be needed then) 

aving staircase and smooth (interpolated) be part of this is that it can be limited if we want to support multiple different interpolation modes.

Reply:
Regarding the interpolation, I agree that it probably makes more sense to separate it from graph display mode.

Don't really understand the PointsMode, Always/Never/Auto? What auto behavior would be?

Reply:
the default uPlot behavior is not to show points when the avg density is too high. this is a major perf benefit, and there's not much use for showing e.g. 10k points in a 600px wide graph.

the current density threshold is 2x point diam.

Reply:
:+1: I'm moving staircase/smooth to its own setting `lineMode`

@dprokop 'auto' is a great point mode that is default in uPlot that grafana has not had.  It will show points when there are few enough that it makes sense.  Ideally we do that always